### PR TITLE
Fix for DaemonRestart [Disruptive] tests

### PR DESCRIPTION
**Pull request by dims**
Fix for https://github.com/kubernetes/kubernetes/issues/126833

- Use correct SSH key injected in CI job to ssh into the nodes (use `GCE_SSH_PRIVATE_KEY_FILE`)
- Kubelet related test was failing because we were using the wrong port (use `KubeletHealthzPort`)
- Controller-Manager and Scheduler related test was failing because we were using `framework.APIAddress()` which is not correct, we should traverse the control plane nodes and pick/use their IP address(es) instead.

#### What type of PR is this?
/kind failing-test
/kind bug
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
You can see the experiments in https://github.com/kubernetes/kubernetes/pull/127005

/test pull-kubernetes-e2e-gce-disruptive-canary

**Comment by dims** [comment]
/hold

( for `pull-kubernetes-e2e-gce-disruptive-canary` to go 🟢 )

/assign @upodroid @ameukam

**Comment by dims** [comment]
cc @hakuna-matatah 

thanks for your help Harish to triage the SSH issue here

**Comment by dims** [comment]
good news all 3 `DaemonRestart [Disruptive]` tests passed https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127033/pull-kubernetes-e2e-gce-disruptive-canary/1829878909965963264

**Comment by hakuna-matatah** [comment]
/lgtm

Could be unit test flakiness.

**Comment by aojea** [comment]
@dims this PR broke these jobs 
- https://testgrid.k8s.io/sig-network-gce#gci-gce-serial-kube-dns
- https://testgrid.k8s.io/sig-network-gce#gci-gce-serial-kube-dns-nodecache
- https://testgrid.k8s.io/sig-network-gce#gci-gce-serial

The problem seems to be in `getControlPlaneNodes()` since it does not find control plane nodes

I0902 06:03:06.901536 10385 daemon_restart.go:317] Unexpected error: 
    <*errors.errorString | 0xc001265e00>: 
    there are currently no ready, schedulable control plane nodes in the cluster
    {
        s: "there are currently no ready, schedulable control plane nodes in the cluster",
    }
[FAILED] there are currently no 
```

> Controller-Manager and Scheduler related test was failing because we were using framework.APIAddress() which is not correct, we should traverse the control plane nodes and pick/use their IP address(es) instead.

we need to make a more solid function, since `node-role` is not universally used, let me send a proposal

File: test/e2e/apps/daemon_restart.go [hunk 1]
```diff
@@ -106,17 +107,20 @@ func (r *RestartDaemonConfig) waitUp(ctx context.Context) {
 	var healthzCheck string
 	if r.enableHTTPS {
 		healthzCheck = fmt.Sprintf(
-			"curl -sk -o %v -I -w \"%%{http_code}\" https://localhost:%v/healthz", nullDev, r.healthzPort)
+			"curl -sk -o %v -I -w \"%%{http_code}\" https://127.0.0.1:%v/healthz", nullDev, r.healthzPort)
```
Comment (aojea) [inline test/e2e/apps/daemon_restart.go:110]: this only works for IPv4, any reason to replace this?

**Comment by aojea** [comment]
I think we can use the taint, the gce nodes on the jobs have the taint but not the label

https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1830403873093718016/artifacts/nodes.yaml

File: test/e2e/apps/daemon_restart.go (hunk 1 above)
Comment (dims) [reply test/e2e/apps/daemon_restart.go:110]: @aojea sorry this sneaked in, will file a PR to revert this line

Comment (aojea) [reply test/e2e/apps/daemon_restart.go:110]: no worries, I submitted another PR https://github.com/kubernetes/kubernetes/pull/127070

**Pull request merged**